core/src/main/java/org/projectnessie/cel/extension/StringsLib.java

```diff
@@ -0,0 +1,340 @@
+            Decls.newFunction(
+                TRIM_SPACE,
+                Decls.newInstanceOverload(
+                    "string_trim", Arrays.asList(Decls.String), Decls.String)),
+            Decls.newFunction(
+                UPPER_ASCII,
+                Decls.newInstanceOverload(
+                    "string_upper_ascii", Arrays.asList(Decls.String), Decls.String)));
+    list.add(option);
+    return list;
+  }
+
+  @Override
+  public List<ProgramOption> getProgramOptions() {
+    List<ProgramOption> list = new ArrayList<>();
+    ProgramOption functions =
+        ProgramOption.functions(
+            Overload.binary(CHAR_AT, Guards.callInStrIntOutStr(StringsLib::charAt)),
+            Overload.binary(INDEX_OF, Guards.callInStrStrOutInt(StringsLib::indexOf)),
+            Overload.binary(LAST_INDEX_OF, Guards.callInStrStrOutInt(StringsLib::lastIndexOf)),
+            Overload.unary(LOWER_ASCII, Guards.callInStrOutStr(StringsLib::lowerASCII)),
+            Overload.function(REPLACE, Guards.callInStrStrStrOutStr(StringsLib::replace)),
+            Overload.binary(SPLIT, Guards.callInStrStrOutStrArr(StringsLib::split)),
+            Overload.binary(SUBSTR, Guards.callInStrIntOutStr(StringsLib::substr)),
+            Overload.unary(TRIM_SPACE, Guards.callInStrOutStr(StringsLib::trimSpace)),
+            Overload.unary(UPPER_ASCII, Guards.callInStrOutStr(StringsLib::upperASCII)));
+    list.add(functions);
+    return list;
+  }
+
+  static String charAt(String str, int index) {
+    if (str.length() == index) {
+      return "";
+    }
+    return String.valueOf(str.charAt(index));
+  }
+
+  static int indexOf(String str, String substr) {
+    return str.indexOf(substr);
+  }
+
+  static int lastIndexOf(String str, String substr) {
+    return str.lastIndexOf(substr);
+  }
+
+  static String lowerASCII(String str) {
+    StringBuilder stringBuilder = new StringBuilder();
+    for (char c : str.toCharArray()) {
+      if (c >= 'A' && c <= 'Z') {
+        stringBuilder.append(Character.toLowerCase(c));
+      } else {
+        stringBuilder.append(c);
+      }
+    }
+    return stringBuilder.toString();
+  }
+
+  static String replace(String str, String old, String replacement) {
+    return str.replace(old, replacement);
+  }
+
+  static String[] split(String str, String separator) {
+    return str.split(Pattern.quote(separator));
+  }
+
+  static String substr(String str, int start) {
+    return str.substring(start);
+  }
+
+  static String trimSpace(String str) {
+    char[] chars = str.toCharArray();
+    int start = 0;
+    int end = str.length() - 1;
+    while (start < str.length()) {
+      if (!UNICODE_WHITE_SPACES.contains(chars[start])) {
+        break;
+      }
+      start++;
+    }
+    while (end > start) {
+      if (!UNICODE_WHITE_SPACES.contains(chars[end])) {
+        break;
+      }
+      end--;
+    }
+
+    return str.substring(start, end + 1);
```

Comment:
Add `try-catch(IndexOutOfBoundsException)` and throw with a non-Java-version dependent message